stages/org.osbuild.dnf4.mark

```diff
@@ -0,0 +1,81 @@
+#!/usr/bin/python3
+"""
+Mark packages in the DNF state database.
+"""
+
+import shutil
+import subprocess
+import sys
+
+from osbuild import api
+
+SCHEMA_2 = """
+"options": {
+  "additionalProperties": false,
+  "properties": {
+    "packages": {
+      "type": "array",
+      "minItems": 1,
+      "description": "Packages and their marks.",
+      "items": {
+        "type": "object",
+        "additionalProperties": false,
+        "required": ["name", "mark"],
+        "properties": {
+          "name": {
+            "type": "string",
+            "description": "Package name."
+          },
+          "mark": {
+            "type": "string",
+            "enum": ["install", "group"],
+            "description": "Package mark."
+          }
+        }
+      }
+    }
+  }
+}
+"""
+
+
+def mark(tree, packages):
+    dnf_bin = shutil.which("dnf-3")
+
+    if not dnf_bin:
+        print("dnf not found")
+        return 1
+
+    markings = {}
+
+    for package in packages:
+        if package["mark"] not in markings:
+            markings[package["mark"]] = []
+
+        markings[package["mark"]] += [package]
```

Comment:
The absolute tiniest of nitpicks:
We could just append `package["name"]` here and then the mark calls bellow will become
```python
  [dnf_bin, "--installroot", tree, "mark", "-y", "group", *markings["group"]]
```

(Really, feel free to ignore this comment)